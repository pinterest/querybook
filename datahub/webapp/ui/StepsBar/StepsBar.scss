.StepsBar {
    padding: 8px;
    .StepsBar-ul {
        counter-reset: step;
        display: flex;
        width: 100%;

        > li {
            list-style-type: none;
            position: relative;
            text-align: center;
            font-size: var(--small-text-size);
            font-weight: bold;

            &:before {
                width: 20px;
                height: 20px;
                content: counter(step);
                counter-increment: step;
                border: 2px solid var(--color-null);
                display: block;
                text-align: center;
                margin: 0 auto 5px auto;
                border-radius: 50%;
                line-height: 18px;

                background-color: var(--color-null);
                color: var(--bg-color);
            }

            &:after {
                height: 2px;
                content: '';
                position: absolute;
                background-color: var(--color-null);
                top: 10px;
                left: calc(-50% + 9px);
                right: calc(50% + 9px);
            }

            &:first-child:after {
                content: none;
            }

            &.StepsBar-active {
                color: var(--color-true);

                &:before {
                    border-color: var(--color-true);
                    background-color: var(--color-true);
                }

                &:after {
                    background-color: var(--color-true);
                }
            }
        }
    }
}
