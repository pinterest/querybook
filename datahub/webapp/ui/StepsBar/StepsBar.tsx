import React from 'react';

import './StepsBar.scss';

interface IProps {
    steps: React.ReactChild[];
    activeStep: number;
}

export const StepsBar: React.FunctionComponent<IProps> = ({
    steps, // Array of strings
    activeStep, // Everything <= activeStep index is active
}) => {
    const widthPerStep = `${(1 / steps.length) * 100}%`;
    const stepsDOM = steps.map((step, index) => (
        <li
            key={index}
            className={index <= activeStep ? 'StepsBar-active' : ''}
            style={{ width: widthPerStep }}
        >
            {step}
        </li>
    ));

    return (
        <div className="StepsBar">
            <ul className="StepsBar-ul">{stepsDOM}</ul>
        </div>
    );
};
