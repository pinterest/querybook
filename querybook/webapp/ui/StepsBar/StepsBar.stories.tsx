import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { StepsBar } from './StepsBar';

const Steps = [
    'one',
    'two',
    'three',
    'four',
    'five',
    'six',
    'seven',
    'eight',
    'nine',
    'ten',
];

export const _StepsBar = (args) => {
    const { numStep, ...otherArgs } = args;
    return (
        <>
            <div style={{ width: '1000px', alignItems: 'stretch' }}>
                <StepsBar steps={Steps.slice(0, numStep)} {...otherArgs} />
            </div>
        </>
    );
};

_StepsBar.args = {
    numStep: 10,
    activeStep: 5,
};
_StepsBar.argTypes = {
    numStep: {
        control: {
            type: 'range',
            min: 1,
            max: 10,
        },
    },
    activeStep: {
        control: {
            type: 'range',
            min: 0,
            max: 9,
        },
    },
};

export default {
    title: 'Stateless/StepsBar',
    decorators: [centered],
};
