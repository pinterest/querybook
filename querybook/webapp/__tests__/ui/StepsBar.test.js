import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { StepsBar } from '../../ui/StepsBar/StepsBar';

it('renders without crashing', () => {
    shallow(<StepsBar steps={[]} activeStep={0} />);
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(<StepsBar steps={[]} activeStep={0} />);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
    it('matches snapshot', () => {
        const wrapper = shallow(
            <StepsBar steps={['one', 'two']} activeStep={0} />
        );
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});
