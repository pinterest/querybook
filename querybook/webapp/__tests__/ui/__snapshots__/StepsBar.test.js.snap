// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot 1`] = `
<div
  className="StepsBar"
>
  <ul
    className="StepsBar-ul"
  />
</div>
`;

exports[`matches enzyme snapshots matches snapshot 2`] = `
<div
  className="StepsBar"
>
  <ul
    className="StepsBar-ul"
  >
    <li
      className="StepsBar-active"
      key="0"
      style={
        Object {
          "width": "50%",
        }
      }
    >
      one
    </li>
    <li
      className=""
      key="1"
      style={
        Object {
          "width": "50%",
        }
      }
    >
      two
    </li>
  </ul>
</div>
`;
